### ISSUE TITLE
Fix race conditions in synchronization

### ISSUE THREAD

**Issue by calvinlfer** [calvinlfer/skull-island#12]
For example:
```javascript
    diskSNIs.map(async sni =>
      await kong.snis.createOrUpdateSNI(sni).catch(err => {
        if (err.statusCode !== 409) console.log(`Error adding SNI`, err.message);
      })
    );
    console.log('SNI updates complete'.green);
    await delay(waitTimeInMs);
```
`diskSNIs.map(...)` returns a list of `Promise`s that are pending. You need to await on the completion of all promises otherwise you cannot guarantee that a request has completed:
```javascript
    const kongCertificatesToUpload = await Promise.all(adjustedKongCertificates);
    const filteredCertificatesToUpload = kongCertificatesToUpload.filter(certificateHasValidPublicAndPrivateKeys);
    const pendingCertResults = filteredCertificatesToUpload.map(async certificate => await kong.certificates.createOrUpdateCertificate(certificate));
    await Promise.all(pendingCertResults);
    console.log('Certificate updates complete'.green);
```

You do it properly here.

**Comment by calvinlfer** [comment]
I have addressed these race conditions in the following hotfix commits:
- https://github.com/calvinlfer/skull-island/commit/96012dddf7d8921645673d01ef4be99797b5037c
- https://github.com/calvinlfer/skull-island/commit/ba3f6e2aa57b038d442fbb1f79cc4ff5f16e287d
- https://github.com/calvinlfer/skull-island/commit/04b2ae766baa7409e19a532c9d1b80c68cef6bcf

I really need a type system to verify my work 😄 

Thanks to @ukayani for pointing this out. 

It seems like these race conditions were periodically causing test cases to fail and I was misinterpreting them incorrectly as Kong taking some time to propagate changes so I have added back Travis-CI integration.

I'll do a release tomorrow so this will be fixed in version >=1.0.14.
